src/main/java/com/sprint/findex_team6/service/IndexService.java

```diff
@@ -0,0 +1,125 @@
+package com.sprint.findex_team6.service;
+
+import com.sprint.findex_team6.dto.IndexInfoDto;
+import com.sprint.findex_team6.dto.request.IndexInfoCreateRequest;
+import com.sprint.findex_team6.dto.request.IndexInfoUpdateRequest;
+import com.sprint.findex_team6.dto.response.ErrorResponse;
+import com.sprint.findex_team6.entity.Index;
+import com.sprint.findex_team6.entity.SourceType;
+import com.sprint.findex_team6.mapper.IndexMapper;
+import com.sprint.findex_team6.repository.IndexRepository;
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+
+@Service
+@RequiredArgsConstructor
+public class IndexService {
+  private IndexMapper indexMapper;
+  private IndexRepository indexRepository;
+
+  public ResponseEntity<?> create(IndexInfoCreateRequest indexInfoCreateRequest){
+
+   if(hasNullFields(indexInfoCreateRequest)){
+     String checkNullField = checkNullField(indexInfoCreateRequest) + "는 필수입니다.";
+     ErrorResponse errorResponse = new ErrorResponse(LocalDateTime.now(),HttpStatus.BAD_REQUEST.value(),"잘못된 요청입니다.", "");
+     return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(errorResponse);
+   }
+
+    String indexClassification = indexInfoCreateRequest.indexClassification();
+    String indexName = indexInfoCreateRequest.indexName();
+    LocalDate baseDate = indexInfoCreateRequest.basePointInTime();
+    BigDecimal baseIndex =indexInfoCreateRequest.baseIndex();
+    Integer employedItemsCount = indexInfoCreateRequest.employedItemsCount();
+    Boolean favorite = indexInfoCreateRequest.favorite();
+
+    Index index = new Index(indexClassification, indexName, employedItemsCount, baseDate, baseIndex,
+        SourceType.USER, favorite);
+    indexRepository.save(index);
+
+    return ResponseEntity.status(HttpStatus.CREATED).body(indexMapper.toDto(index));
+  }
+
+  public IndexInfoDto update(IndexInfoUpdateRequest request, Long id){
+    Integer employedItemsCount = request.employedItemsCount();
+    LocalDate basePointInTime = request.basePointInTime();
+    BigDecimal baseIndex = request.baseIndex();
+    Boolean favorite = request.favorite();
+
+    Index index = null;
+    if(indexRepository.findById(id).isPresent()){
+      index = indexRepository.findById(id).get();
+    }
+    else{
+      throw new NoSuchElementException();
+    }
```

Comment:
```suggestion
    Index index = indexRepository.findById(id)
    .orElseThrow(() -> new NoSuchElementException())
```
orElseThrow를 사용하면 더 간결하게 표현할 수 있어요.